### DISCUSSION TITLE
Is it possible to remove values from context?

### DISCUSSION THREAD

**Discussion by bryanjswift** [statelyai/xstate#3071] [Q&A]
I have a context type that is a union of two states. One of them has a property and the other does not, let's call them failure and success. Previously, if a failure happened after a success had happened I would return the full context from an `assign` action and the property that only exists in the success state would be gone (matching my types). It appears that updating to v4.30.1 of xstate from v4.27.0 has changed this functionality. Was the change a bug fix or an unintended consequence of a performance improvement?

Is there an alternative way to remove a value from context?

**Comment by Andarist** [comment]
Could you prepare a repro case? I don't see how `assign(context => context)` would **ever** change the current value of the context. I assume that you might be talking about a slightly different case but without taking a look at the code I can't really comment on that.

**Comment by bryanjswift** [comment]
I stripped my package down a lot but probably not as minimal as it could be. The test that fails is on line 124 of `test/AccessCodeMachine.test.ts`. The "action" that should be clearing out a value from context is `clearAttendeeId` on line 88 of `src/AccessCodeMachine.ts`.

[xstate-context-repro.zip](https://github.com/statelyai/xstate/files/8118316/xstate-context-repro.zip)
